File: SecondYear/ApoorvaSri123_ApoorvaSrivastava_2226it1116_2/Week_3/binary_search.c
#include <stdio.h>

int binarySearch(int arr[], int n, int num) {
    int left = 0;
    int right = n - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == num) {
            return mid;
        } else if (arr[mid] < num) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return -1;
}

int main() {
    int n;
    printf("Enter the size of the array: ");
    scanf("%d", &n);

    int arr[n];
    
    printf("Enter the elements of the array in sorted order:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int numToFind;
    printf("Enter the number to find: ");
    scanf("%d", &numToFind);

    int result = binarySearch(arr, n, numToFind);

    if (result != -1) {
        printf("Number found at index: %d\n", result);
    } else {
        printf("-1\n");
    }

    return 0;
}

File: SecondYear/Divyansh670_DivyanshSrivastav_2226it1029_2/Week_3/binary_search.java
import java.util.Arrays;
import java.util.Scanner;

public class binary_search {
    public static int binarySearch(int[] arr, int num) {
        int left = 0;
        int right = arr.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (arr[mid] == num) {
                return mid; 
            } else if (arr[mid] < num) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return -1; 
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        
        int n = scanner.nextInt();
        int[] arr = new int[n];

        
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }
        Arrays.sort(arr); 

        
        int numToFind = scanner.nextInt();

        int result = binarySearch(arr, numToFind);

        if (result != -1) {
            System.out.println(result);
        } else {
            System.out.println("-1");
        }

        scanner.close();
    }
}
